test/SILOptimizer/string_switch.swift

```diff
@@ -0,0 +1,137 @@
+// RUN: %target-swift-frontend -primary-file %s -O -Xllvm -sil-disable-pass=FunctionSignatureOpts -module-name=test -emit-sil -enforce-exclusivity=unchecked | %FileCheck %s
```

Comment:
It might make sense to add a few more tests with more complicated logic/branches.